packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/distant-table/distant-table.service.ts

```diff
@@ -23,43 +23,42 @@ export class DistantTableService {
     >,
   ) {}
 
-  public getDistantTableColumns(
-    remoteServer: RemoteServerEntity<RemoteServerType>,
-    tableName: string,
-  ): PostgresTableSchemaColumn[] {
-    if (!remoteServer.availableTables) {
-      throw new BadRequestException(
-        'Remote server available tables are not defined',
-      );
-    }
-
-    return remoteServer.availableTables[tableName];
-  }
-
   public async fetchDistantTables(
     remoteServer: RemoteServerEntity<RemoteServerType>,
     workspaceId: string,
   ): Promise<DistantTables> {
-    return this.createAvailableTables(remoteServer, workspaceId);
+    if (remoteServer.schema) {
+      return this.getDistantTablesFromDynamicSchema(remoteServer, workspaceId);
+    }
+
+    return this.getDistantTablesFromStaticSchema(remoteServer);
   }
 
-  private async createAvailableTables(
+  public async getDistantTableColumns(
     remoteServer: RemoteServerEntity<RemoteServerType>,
     workspaceId: string,
-  ): Promise<DistantTables> {
+    tableName: string,
+  ): Promise<PostgresTableSchemaColumn[]> {
     if (remoteServer.schema) {
```

Comment:
nit: Could be refactored like this if you want, also distantTables[tableName] could be undefined right? What do we want to do in this case? Should we throw or simply return an empty array?
```typescript
public async getDistantTableColumns(
  remoteServer: RemoteServerEntity<RemoteServerType>,
  workspaceId: string,
  tableName: string,
): Promise<PostgresTableSchemaColumn[]> {
  const distantTables = remoteServer.schema 
    ? await this.getDistantTablesFromDynamicSchema(remoteServer, workspaceId, tableName)
    : await this.getDistantTablesFromStaticSchema(remoteServer);

  return distantTables[tableName] || [];
}
```

Reply:
Even better, you can actually reuse `this.fetchDistantTables` above to get the distantTables I think, code looks similar.

Reply:
Right, I updated this.fetchDistantTables so that it also takes in an optional tableName to only fetch that table (we don't want to fetch all tables here)

You are right, I handled the undefined distantTables[tableName] case during the update but not at the sync which would cause an error. I will update that. (It should not throw an error because it could be that the distant table was deleted which is a possibliity)